pandas/tests/io/formats/test_to_csv.py

```diff
@@ -300,3 +300,19 @@ def test_to_csv_stdout_file(self):
         output = sys.stdout.getvalue()
         assert output == expected_ascii
         assert not sys.stdout.closed
+
+    def test_to_csv_write_to_open_file(self):
+        # GH 21696
+        df = pd.DataFrame({'a': ['x', 'y', 'z']})
+        expected = '''\
+manual header
+x
+y
+z
+'''
+        with tm.ensure_clean('test.txt') as path:
```

Comment:
can you parameterize this with flush (both on and off), or does this matter?

Reply:
I don't think this matters.